unstructured/utils.py

```diff
@@ -280,3 +301,105 @@ def scarf_analytics():
                 )
     except Exception:
         pass
+
+
+def draw_bboxes_on_pdf_or_image(
+    file_path,
+    elements,
+    desired_width=20,
+    save_images=False,
+    save_coordinates=False,
+    output_folder=None,
+    plot=False,
+):
```

Comment:
Some general thoughts:
1. We definitely should not have to pick up `matplotlib` as a base level dependency. With that in mind this should probably be moved somewhere else like a `visualize.py` file, and the import should be moved inside the function. The tests should live in their own file as well.
1. I think this function does too much. At the very least, the logic of plotting should have its own function with inputs only an object representing a single image and an object representing a single bounding box, and output should be an object representing a single image on which the box has been plotted. The business of loading and saving the image, resizing the image, and plotting multiple boxes across multiple pages are separate concerns. Then we can have a function of plotting multiple boxes to a single page/image, that takes the minimum necessary parameters and calls the single image/box function. It's okay to have a one-stop-shop function like this, but if I already have an image in memory that I want to annotate I shouldn't have to write a new function.
1. We need the normal code quality perks like type hints.
1. I prefer the pattern of plotting directly to the `PIL` `Image` and returning that object rather than using a heavier dependency like `matplotlib`. I could potentially live with `matplotlib`, but this function seems to be written around using a notebook with calls like `plt.show()`, where using only `PIL` is notebook friendly while also usable in e.g. a script. In any case it's nicer to return an object with a notebook-friendly repr method than to have a `plot` parameter. `PIL.Image.Image` satisfies this, and so does `matplotlib.figure.Figure` IIRC.
1. Couldn't we drop the `save_images` parameter and assume anything other than `None` in `output_folder` indicates the user wants to save the images? I realize using both parameters is a pattern we've used elsewhere but it doesn't seem ideal.
1. I'm not sure `save_coordinates` belongs here. Probably it was relevant to your use case, so I'm open minded, but we have existing functions that output JSON with coordinates, and my feeling is that it would be better to incorporate those with this or vice-versa.
1. I'm not sure about the resizing functionality here either. Again, you were probably using this, but it's like we're opening the door to a function that supports full fledged image formatting. If we output an image or figure, then all that functionality is inherent in the output object itself, which seems good enough to me.